environment_mac.yml

```diff
@@ -0,0 +1,44 @@
+name: generic_expression_mac
```

Comment:
Is the difference between the mac and linux environment files just that the mac env leaves out the gxx dependency? If that's the case it may be easier to use a single environment and just not list gxx since your compiler will change between operating systems and (maybe) processors anyway. 

If there's a reason that you are setting gxx explicitly though then the solution you're using is probably the best one

Reply:
This was another legacy package that I have, so I'm not sure if its required. I'll try to test this -- yay for github actions.